### ISSUE TITLE
UNCServerName set to NULL

### ISSUE THREAD

**Issue by 3lp4tr0n** [evilashz/CheeseOunce#2]
Two questions/issues with the python script:

1. How is the DCE request supposed to target the server if **request['UNCServerName'] = NULL** ? (line 47)
2. When adding the target IP address  **request['UNCServerName'] = '%s\x00' % self.machine** (line 47) there is no authentication. Are you able to replicate this?

**Comment by eversinc33** [comment]
Hey, I added the python poc today. I tested it in my lab and worked fine. Did you have any issues replicating it?

Regarding both your issues: the logic lies in `connect()` where the string binding (`r'ncacn_np:%s[\PIPE\eventlog]' % self.machine`) targets the server: `rpctransport = transport.DCERPCTransportFactory(self.stringBinding)` and authentication happens with `rpctransport.set_credentials(self.username, self.password, self.domain, lmhash, nthash)`.
